File: springwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/reactive/QueueSubscription.java
package com.qiusuo.webfluxtutorial.reactive;

import org.reactivestreams.Subscriber;
import org.reactivestreams.Subscription;

import java.util.LinkedList;
import java.util.Queue;

public class QueueSubscription implements Subscription {
    private Queue<String> queuedEntities;
    private Subscription parent;
    private Subscriber<String> subscriber;

    public QueueSubscription(Subscriber subscriber) {
        queuedEntities = new LinkedList<>();
    }
    @Override
    public void request(long l) {
        parent.request(l);
    }

    @Override
    public void cancel() {

    }
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/reactive/QueueSubscription.kt
package com.qiusuo.webfluxtutorial.reactive

import org.reactivestreams.Subscriber
import org.reactivestreams.Subscription
import java.util.*

class QueueSubscription(subscriber: Subscriber<*>?) : Subscription {
    private val queuedEntities: Queue<String>
    private val parent: Subscription? = null
    private val subscriber: Subscriber<String>? = null
    override fun request(l: Long) {
        parent!!.request(l)
    }

    override fun cancel() {}

    init {
        queuedEntities = LinkedList()
    }
}
